### PR TITLE
added recursive check for old consentRecordHashes

### LINKED ISSUE #36: Metadata.consentRecordHash must be unique over all attachments
This is to prevent replaying older consent records.
It has to be enforced in the merge flow

### PR DESCRIPTION
**Pull request by woutslakhorst**
- [x] Create test showing new check works

Fixes #36 

Signed-off-by: Wout Slakhorst <wout.slakhorst@nedap.com>

### CODE REVIEW

File: flows/src/main/kotlin/nl/nuts/consent/flow/ConsentFlows.kt [hunk 1]
```diff
@@ -55,24 +57,48 @@ object ConsentFlows {
...
     /**
      * A check for duplicate attachments, this is done outside the contract because the contents of the attachments is required
+     * It recursively checks previous transactions as well
      */
-    fun raiseOnDuplicateConsentRecords(consentState: ConsentState, consentBranch: ConsentBranch, serviceHub: ServiceHub, side:String = "origin") {
-        val consentRecordHashInState = containedConsentRecordHash(consentState, serviceHub)
-        val consentRecordHashInBranch = containedConsentRecordHash(consentBranch, serviceHub)
+    fun raiseOnDuplicateConsentRecords(stateRef: StateRef, attachments: Set<SecureHash>, serviceHub: ServiceHub, side:String = "origin") {
+        // new hashes
+        val newHashes = containedConsentRecordHash(attachments, serviceHub)
```
Comment (stevenvegt) [inline flows/src/main/kotlin/nl/nuts/consent/flow/ConsentFlows.kt:80]: How expensive is this? Can it be a parameter for this method so it runs only once?

**Review by stevenvegt — COMMENTED** [review]
What if I update a record and set an enddate and later remove it again. Does it fail?

**Comment by woutslakhorst** [comment]
No, since the proof for that part is in the consent proof: document with wet signature or some signed structure

File: flows/src/main/kotlin/nl/nuts/consent/flow/ConsentFlows.kt (hunk 1 above)
Comment (woutslakhorst) [reply flows/src/main/kotlin/nl/nuts/consent/flow/ConsentFlows.kt:80]: it has to fetch transaction -> states -> transaction -> etc, each being a db call.

**Pull request merged**